docs/guides/isolated_systems.md

```diff
@@ -0,0 +1,221 @@
+# Isolated systems guide
+
+SQLMesh assumes that production and non-production data systems can access one another, but that is not always the case.
+
+Some architectures isolate systems from one another, preventing production data from being accessed in non-production systems. This isolation is usually motivated by information security concerns, as non-production systems may be accessible to more users and/or have looser security restrictions.
+
+This guide explains how to use SQLMesh with isolated systems. It also clarifies how isolating systems affects SQLMesh's behavior and why the systems still enjoy the benefits of many SQLMesh features.
+
+## Terminology
+
+Isolated systems are sometimes referred to as "isolated environments," but that term is ambiguous.
+
+In this context, the things being isolated are servers and databases, and the isolation is enforced by limiting network communications between the systems. We will refer to these as "isolated systems" - the "production system" and "non-production system."
```

Comment:
this reads like a technical paper :). can we make it a bit more relatable like two separate data warehouses that can't talk to each other?